server/commands/teamUpdater.ts

```diff
@@ -58,10 +61,47 @@ const teamUpdater = async ({ params, user, team, ip }: TeamUpdaterProps) => {
   if (inviteRequired !== undefined) {
     team.inviteRequired = inviteRequired;
   }
+  if (allowedDomains !== undefined) {
+    const existingAllowedDomains = await TeamDomain.findAll({
+      where: { teamId: team.id },
+      transaction,
+    });
 
-  const changes = team.changed();
+    // Only keep existing domains if they are still in the list of allowed domains
+    const newAllowedDomains = team.allowedDomains.filter((existingTeamDomain) =>
+      allowedDomains.includes(existingTeamDomain.name)
+    );
 
-  const transaction: Transaction = await sequelize.transaction();
+    // Add new domains
+    const existingDomains = team.allowedDomains.map((x) => x.name);
+    const newDomains = allowedDomains.filter(
+      (newDomain) => newDomain !== "" && !existingDomains.includes(newDomain)
+    );
+    for (const newDomain of newDomains) {
+      newAllowedDomains.push(
+        await TeamDomain.create(
```

Comment:
This could be a `await Promise.all(newDomains.map(` type of thing to parallelize the queries